examples/tv-app/tv-common/clusters/messages/MessagesManager.cpp

```diff
@@ -18,39 +18,152 @@
 #include "MessagesManager.h"
 
 #include <app-common/zap-generated/attributes/Accessors.h>
+#include <vector>
 
 using namespace std;
+using namespace chip;
 using namespace chip::app;
 using namespace chip::app::Clusters::Messages;
 using Message = chip::app::Clusters::Messages::Structs::MessageStruct::Type;
 
 // Commands
-void MessagesManager::HandlePresentMessagesRequest(
-    const chip::ByteSpan & messageId, const MessagePriorityEnum & priority,
-    const chip::BitMask<MessageControlBitmap> & messageControl, const chip::app::DataModel::Nullable<uint32_t> & startTime,
-    const chip::app::DataModel::Nullable<uint16_t> & duration, const chip::CharSpan & messageText,
-    const chip::Optional<chip::app::DataModel::DecodableList<MessageResponseOption>> & responses)
+void MessagesManager::HandlePresentMessagesRequest(const ByteSpan & messageId, const MessagePriorityEnum & priority,
+                                                   const BitMask<MessageControlBitmap> & messageControl,
+                                                   const DataModel::Nullable<uint32_t> & startTime,
+                                                   const DataModel::Nullable<uint16_t> & duration, const CharSpan & messageText,
+                                                   const Optional<DataModel::DecodableList<MessageResponseOption>> & responses)
 {
-    Message message{
-        // TODO: Enable id
-        chip::ByteSpan(), priority, messageControl, startTime, duration,
-        // TODO: Enable text
-        chip::CharSpan()
-        // TODO: Convert responses to Optional<chip::app::DataModel::List<const
-        // chip::app::Clusters::Messages::Structs::MessageResponseOptionStruct::Type>> message.responses = responses;
-    };
-
-    mMessages.push_back(message);
+    std::vector<char *> needToFree;
+    std::vector<MessageResponseOption *> optionToFree;
+
+    uint8_t * messageIdBuffer = nullptr;
+    char * messageTextBuffer  = new char[messageText.size()];
+    VerifyOrExit(messageTextBuffer != nullptr,
+                 ChipLogProgress(Controller, "HandlePresentMessagesRequest messageTextBuffer alloc failed"));
+    needToFree.push_back(messageTextBuffer);
+    memcpy(messageTextBuffer, messageText.data(), messageText.size());
+
+    ChipLogProgress(Controller, "HandlePresentMessagesRequest message:%s size:%lu", messageTextBuffer, messageText.size());
+
+    messageIdBuffer = new uint8_t[messageId.size()];
```

Comment:
why are you copying things here? Why is the span not usable as-is?

This should be a unique_ptr

Reply:
the contents of the span gets freed after the method invocation but the new message needs to persist